src/utils/ucc_math.h

```diff
@@ -12,4 +12,26 @@
 #define ucc_min(_a, _b) ucs_min((_a), (_b))
 #define ucc_max(_a, _b) ucs_max((_a), (_b))
 
+static inline uint32_t ucc_ep_map_eval(ucc_ep_map_t map, uint32_t rank)
+{
+    uint32_t r;
+    switch(map.type) {
+    case UCC_EP_MAP_FULL:
+        r = rank;
+        break;
+    case UCC_EP_MAP_STRIDED:
+        r = map.strided.start + rank*map.strided.stride;
+        break;
+    case UCC_EP_MAP_ARRAY:
+        r = *((uint32_t*)((ptrdiff_t)map.array.map + rank*map.array.elem_size));
+        break;
+    case UCC_EP_MAP_CB:
+        r = (uint32_t)map.cb.cb(rank, map.cb.cb_ctx);
+        break;
+    default:
+        r = -1;
```

Comment:
function returns uint32_t and -1 looks confusing here. I would better use some constant